crates/compiler/gen_llvm/src/llvm/lowlevel.rs

```diff
@@ -1169,6 +1169,135 @@ pub(crate) fn run_low_level<'a, 'ctx>(
+
+            arguments_with_layouts!((arg, arg_layout));
+            //Get the actual value that we are converting to
+            let result = match layout_interner.get_repr(arg_layout) {
+                //If integer
+                LayoutRepr::Builtin(Builtin::Int(width)) => {
+                    //Extend to I128
+                    let Ok(converted_value) = (if width.is_signed() {
+                        env.builder
+                            .build_int_s_extend(arg.into_int_value(), to, "inc_cast")
+                    } else {
+                        env.builder
+                            .build_int_z_extend(arg.into_int_value(), to, "inc_cast")
+                    }) else {
+                        todo!()
+                    };
+                    //Add decimal places
+                    let Ok(result) =
+                        env.builder
+                            .build_int_mul(converted_value, llvm_digits, "deci_cast")
+                    else {
+                        todo!()
+                    };
+                    result.as_basic_value_enum()
+                }
+                LayoutRepr::Builtin(Builtin::Float(_)) => {
+                    let Ok(float_result) = env.builder.build_float_mul(
+                        arg.into_float_value(),
+                        llvm_divisor,
+                        "decf_cast",
+                    ) else {
+                        todo!()
+                    };
+                    let Ok(int_result) =
+                        env.builder
+                            .build_float_to_signed_int(float_result, to, "dec_place")
+                    else {
+                        todo!()
+                    };
+                    int_result.as_basic_value_enum()
+                }
+                LayoutRepr::Builtin(Builtin::Decimal) => arg,
+                _ => {
+                    todo!()
+                }
+            };
+            //Look and see if the conversion is checked or not
+            if checked {
+                let bool_false = env.context.bool_type().const_zero();
+                let with_overflow = match layout_interner.get_repr(arg_layout) {
+                    //If the value being is converted into a integer
+                    LayoutRepr::Builtin(Builtin::Int(width)) => {
+                        //`Dec` can always fit `I64`s and `U64`s and smaller, no need to check at runtime
+                        if width.stack_size() <= 4 {
+                            bool_false
+                        } else if width.is_signed() {
+                            let compare_arg = env
+                                .builder
+                                .build_int_z_extend_or_bit_cast(
+                                    arg.into_int_value(),
+                                    to,
+                                    "int_cast",
+                                )
+                                .unwrap();
```

Comment:
Can this unwrap be reasonably avoided so we don't crash the compiler with a bad error message?